mypy/options.py

```diff
@@ -119,7 +120,7 @@ def __init__(self) -> None:
         self.plugins = []  # type: List[str]
 
         # Per-module options (raw)
-        self.per_module_options = {}  # type: Dict[Pattern[str], Dict[str, object]]
+        self.per_module_options = collections.OrderedDict()  # type: MutableMapping[Pattern[str], Dict[str, object]]
```

Comment:
I think we should consider making this of type `OrderedDict`, because that's important for the semantics.